foyer/forcefield.py

```diff
@@ -51,6 +52,34 @@ def preprocess_forcefield_files(forcefield_files=None):
         xml_contents = re.sub(r"(def\w*=\w*[\"\'])(.*)([\"\'])", lambda m: m.group(1) + re.sub(r"&(?!amp;)", r"&amp;", m.group(2)) + m.group(3),
                               xml_contents)
 
+        try:
+            '''
+            Sort topology objects by precedence, defined by the number of
+            `type` attributes specified, where a `type` attribute indicates
+            increased specificity as opposed to use of `class`
+            '''
+            root = ET.fromstring(xml_contents)
+            for element in root:
+                if 'Force' in element.tag:
+                    element[:] = sorted(element, key=lambda child: -1 * len([attr_name for attr_name in child.keys() if 'type' in attr_name]))
+            xml_contents = ET.tostring(root, method='xml')
+            '''
+            Convert the output byte string to unicode. This method is specific to
+            the Python version (2 or 3)
+            '''
+            try:
+                xml_contents = unicode(xml_contents, 'utf-8')
+            except NameError:
+                xml_contents = str(xml_contents, 'utf-8')
```

Comment:
I think you can get rid of this and simply write `xml_contents = ET.tostring(root, method='xml').decode()`